src/validation.cpp

```diff
@@ -1695,15 +1695,15 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package,
         return PackageMempoolAcceptResult(package_state_quit_early, {});
     }
 
-    // IsChildWithParents() guarantees the package is > 1 transactions.
-    assert(package.size() > 1);
```

Comment:
Given that this assert exists, I'm worried we have other places with very strict assumptions on what `IsChildWithParents` can guarantee. Perhaps it's safer to just have the RPC code or `ProcessNewPackage` to direct a single transaction to ATMP instead.

Reply:
I'd rather not; we almost immediately handle singletons via `AcceptSubPackage` internally, I'm not convinced this is more risky than adding more complexity on top.

Reply:
This is what testmempoolaccept does